src/controller/CommissionerDiscoveryController.h

```diff
@@ -235,6 +235,21 @@ class DLL_EXPORT AppInstallationService
      */
     virtual bool LookupTargetContentApp(uint16_t vendorId, uint16_t productId) = 0;
 
+    /**
+     * @brief
+     *   Called to get app's installation error code.
+     *
+     * CdError status is designed for CommissionerDeclaration and should be set by using SetErrorCode() and
+     * sent back to the client as a CDC Message. It is expected that app during installation can return following errors:
+     * kNoError, kAppInstallConsentPending, kAppInstalling, kAppInstallFailed, kAppInstalledRetryNeeded
```

Comment:
If this is a subset of errors, the comment should explain why only this subset will be returned here (and that it is a subset).

If this is an exhaustive set, then we should NOT have this list as this would mean we have to maintain the list in sync. 

Overall listing anything except kNoError or special one-offs adds a maintenance burden to update the comment if the underlying code changes (and worse, you return a generic CdError yet your api claims to never return the entire set of errors and if clients rely on that, they may break on updates).

How about something like:

```
Returns a subset of CDError that are specific during application instalation lifecycle. In particular:
  - kNoError means no error
  - kAppInstallConsentPending is returned if no application is available
  - other kApp* constants for lifecycle managemnt (kAppInstalling, kAppInstallFailed, kAppInstallRetryNeeded)
  - other CdError constants are EXPLICITLY DISALLOWED
```

unsure if the above comment is true ... however please see how it can be phrased to make expectations clear. Is for example `kUnexpectedCommissionerPasscodeReady` (random pick) possible here? it is a valid CdError value. How about `kConfigurationFailed`?

Should we maybe consider having a separate enum that is more restricted and that can be directly converted to a CdError instead?